src/function/scalar/string/contains.cpp

```diff
@@ -151,15 +152,39 @@ struct ContainsOperator {
 	}
 };
 
-ScalarFunction ContainsFun::GetFunction() {
-	return ScalarFunction("contains",                                   // name of the function
-	                      {LogicalType::VARCHAR, LogicalType::VARCHAR}, // argument list
-	                      LogicalType::BOOLEAN,                         // return type
-	                      ScalarFunction::BinaryFunction<string_t, string_t, bool, ContainsOperator>);
+static unique_ptr<FunctionData> ListContainsBind(ClientContext &context, ScalarFunction &bound_function,
+                                                 vector<unique_ptr<Expression>> &arguments) {
+	return ListContainsOrPositionBind<LogicalType::BOOLEAN>(context, bound_function, arguments);
+}
+
+static void ListContainsFunction(DataChunk &args, ExpressionState &state, Vector &result) {
+	(void)state;
+	return ListContainsOrPosition<bool, ContainsFunctor, ListArgFunctor>(args, result);
+}
+
+static ScalarFunction GetListFunction() {
+	return ScalarFunction({LogicalType::LIST(LogicalType::ANY), LogicalType::ANY}, LogicalType::BOOLEAN,
+	                      ListContainsFunction, ListContainsBind, nullptr);
+}
+
+ScalarFunctionSet ContainsFun::GetFunctions() {
+	ScalarFunction string_fun("contains", {LogicalType::VARCHAR, LogicalType::VARCHAR}, LogicalType::BOOLEAN,
+	                          ScalarFunction::BinaryFunction<string_t, string_t, bool, ContainsOperator>);
+	auto list_fun = GetListFunction();
+	ScalarFunctionSet set("contains");
+	set.AddFunction(string_fun);
+	set.AddFunction(list_fun);
+	return set;
+}
+
+ScalarFunction ContainsFun::GetStringContains(ClientContext &context) {
+	auto set = ContainsFun::GetFunctions();
+	return set.GetFunctionByArguments(context, {LogicalType::VARCHAR, LogicalType::VARCHAR});
```

Comment:
Binding for this seems excessive - can we move the `ScalarFunction string_fun` into this function and instead call it from within `ContainsFun::GetFunctions`? That way we also don't need a client context here.